src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/SelfRefreshingKeySet.java

```diff
@@ -143,11 +143,11 @@ private JsonWebKey getKeySimple(String kid) {
             }
 
         } else {
-            return jsonWebKeys.getKey(kid);
+            return jsonWebKeys.getKeyByKeyId(kid);
         }
     }
 
-    private synchronized JsonWebKey waitForRefreshToFinish(String kid) {
+    private synchronized JWK waitForRefreshToFinish(String kid) {
```

Comment:
same here? is it a global object? if it creates per request there is no need to synchronize it